components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2803,6 +2803,44 @@ public IdentityProvider getIdPByRealmId(String realmId, int tenantId, String ten
 
     }
 
+    /**
+     * Get the enabled IDP of the given realm id.
+     *
+     * @param realmId       Realm ID of the required identity provider.
+     * @param tenantId      Tenant ID of the required identity provider.
+     * @param tenantDomain  Tenant domain of the required identity provider.
+     * @return              Enabled identity provider of the given realm id.
+     * @throws IdentityProviderManagementException Error when getting the identity provider.
+     * @throws SQLException                        Error when executing SQL query.
+     */
+    public IdentityProvider getEnabledIdPByRealmId(String realmId, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+        String idPName = null;
+
+        try {
+            String sqlStmt = IdPManagementConstants.SQLQueries.GET_ENABLED_IDP_NAME_BY_REALM_ID_SQL;
+            prepStmt = dbConnection.prepareStatement(sqlStmt);
+            prepStmt.setInt(1, tenantId);
+            prepStmt.setInt(2, MultitenantConstants.SUPER_TENANT_ID);
+            prepStmt.setString(3, realmId);
+            prepStmt.setString(4, IdPManagementConstants.IS_TRUE_VALUE);
+            rs = prepStmt.executeQuery();
+            if (rs.next()) {
+                idPName = rs.getString("NAME");
```

Comment:
use constant for "Name"

Reply:
It's been used in this way for all the methods throughout the IdPManagementDAO class, and no constants have been defined for such variables throughout the class.

So I think it's better to keep it this way.

Reply:
Fixed and included in [bcbb83d](https://github.com/wso2/carbon-identity-framework/pull/4991/commits/bcbb83da651df7827aece0ed28b33f5166e59e0f)